File: Payhub - SDK/ruby_SDK/PayHubSDK/lib/PayHubSDK/com/payhub/ws/model/void_transaction.rb
class VoidTransaction < WsConnections
  include JsonSerializer
  ATTRS=[:transaction_id, :merchant]
  attr_accessor :url_basePath,*ATTRS
  @url=""
  VOID_ID_LINK = "void/"

  def initialize (merchant, transaction_id)
    Util::validate_params(self.class.name, merchant, transaction_id)
    @merchant=merchant
    @transaction_id=transaction_id
  end

  def url=(str)
    @url_basePath = str
  end

  def url
    return @url_basePath+VOID_ID_LINK
  end

  def performVoidTransaction(http,request)
    json = self.serialize_to_json
    request.body = json
    response = JSON.parse(doPost(http,request))
    result=VoidResponseInformation.new
    if not response.include?('errors')
      result=VoidResponseInformation.from_json(JSON.generate(response))
    else
      errors ||= Array.new
      response['errors'].each do |error|
        errors_aux=Errors.from_json(JSON.generate(error))
        errors.push(errors_aux)
      end
      result.errors=errors
    end
    return result
  end
end

File: Payhub - SDK/php_SDK/com/payhub/ws/model/VoidTransaction.php
<?php

/**
 * Created by PhpStorm.
 * User: agustin
 * Date: 28/07/2015
 * Time: 09:03
 */
class VoidTransaction extends WsConnections
{
    private $url_basePath;
    private $url;
    public $merchant;
    public $transaction_id;
    public static $VOID_ID_LINK = "void/";

    /**
     * VoidTransaction constructor.
     * @param $merchant
     * @param $transaction_id
     */
    public function __construct($merchant, $transaction_id)
    {
        $this->merchant = $merchant;
        $this->transaction_id = $transaction_id;
    }


    /**
     * @return mixed
     */
    public function getUrlBasePath()
    {
        return $this->url_basePath;
    }

    /**
     * @param mixed $url_basePath
     */
    public function setUrlBasePath($url_basePath)
    {
        $this->url_basePath = $url_basePath;
    }

    /**
     * @return mixed
     */
    public function getUrl()
    {
        return $this->url_basePath.'void/';
    }

    /**
     * @param mixed $url
     */
    public function setUrl($url)
    {
        $this->url_basePath = $url;
    }

    /**
     * @return mixed
     */
    public function getMerchant()
    {
        return $this->merchant;
    }

    /**
     * @param mixed $merchant
     */
    public function setMerchant($merchant)
    {
        $this->merchant = $merchant;
    }

    /**
     * @return mixed
     */
    public function getTransactionId()
    {
        return $this->transaction_id;
    }

    /**
     * @param mixed $transaction_id
     */
    public function setTransactionId($transaction_id)
    {
        $this->transaction_id = $transaction_id;
    }
    private function object_unset_nulls($obj)
    {
        if(is_null($obj)){$obj=$this;}
        $arrObj = is_object($obj) ? get_object_vars($obj) : $obj;
        foreach($arrObj as $key => $val)
        {
            $val = (is_array($val) || is_object($val)) ? $this->object_unset_nulls($val) : $val;
            if (is_array($obj))
                $obj[$key] = $val;
            else
                $obj->$key = $val;
            if($val == null)
                unset($obj->$key);
        }
        return $obj;
    }
    public function performVoidTransaction($request){
        $json = json_encode($this->object_unset_nulls(null));
        echo $json;
        curl_setopt($request, CURLOPT_POSTFIELDS, $json);
        $result = $this->doPost($request,$this->getUrl());
        $response = VoidResponseInformation::fromArray($result);
        return $response;
    }
}